### ISSUE TITLE
Checking typeof window.CustomEvent

### ISSUE THREAD

**Issue by alexandrethsilva** [Stanko/window-scroll-manager#2]
Hi @Stanko, first of all, thanks for working on this lib.

I'm currently using another library of yours (`react-window-decorators`), which has a direct dependency on this one here.

While testing on an iPad 2 (arguably old, but still) anything triggering a `CustomEvent` crashed. Upon inspection, I realised two things.

The specification signature of the method is as follows, with the first argument being obligatory and all others being optional:
```
[Constructor(DOMString type, optional CustomEventInit eventInitDict),
 Exposed=(Window,Worker)]
interface CustomEvent : Event {
  readonly attribute any detail;

  void initCustomEvent(DOMString type, optional boolean bubbles = false, optional boolean cancelable = false, optional any detail = null);
};

dictionary CustomEventInit : EventInit {
  any detail = null;
};
```

In comparison to the spec, your signature goes for a different signature, where you take an argument denominated `userParams` and pulls each of the options from it.

```javascript
if (typeof window !== 'undefined' && typeof window.CustomEvent !== 'function') {
  var CustomEventPollyfill = function CustomEventPollyfill(event, userParams) {
    var params = {
      bubbles: userParams.bubbles || false,
      cancelable: userParams.cancelable || false,
      detail: userParams.detail || undefined
    };
    var evt = document.createEvent('CustomEvent');
    evt.initCustomEvent(event, params.bubbles, params.cancelable, params.detail);
    return evt;
  };

  CustomEventPollyfill.prototype = window.Event.prototype;
  window.CustomEvent = CustomEventPollyfill;
}
```

My first question here is whether this is really the ideal approach? Being a polyfill, shouldn't the interface be kept the same for the sake of compatibility?

The second thing, which is the main trigger of my issue really, is that `window.CustomEvent` is being replaced by your implementation even when being available, since the check you've implemented looks the its type being a `"function"`, but not `"object"`. In iOS 9 Safari, `typeof window.CustomEvent` is `"object"` and that leads it to being overwritten.

As you can see, it seems that these two things are coming together and crashing. In my specific case there's a synthetic event coming from a Bootstrap slider, which uses the spec interface, but then crashes when running against yours.

I'd be happy to open a PR with the 2 changes for this (they are very simple), but I reckon it makes it a breaking change (even though internal since you don't expose this to the library users) and wanted to ask your take on it before. For me it's being quite a headache currently as I can't deploy a iOS9-compatible version of my app without removing `react-window-decorators` because of this issue.

Long story short, can I go just go ahead and open the PR?

**Comment by Stanko** [comment]
Hello @alexandrethsilva,
First, thank you so much for the elaborate description of the problem. I wasn't aware that polyfill signature is different than the real thing (guilty as charged). I didn't want to add a external polyfill, as it would involve adding a build system (to resolve dependencies), so I found this one 🙈 

Of course you can open a pr, that would greatly appreciated!

As soon as I merge the fix, I'll apply it here as well:
https://github.com/Stanko/react-window-decorators/blob/6f7509eec8a861ceccae26156230705577d6a5d5/src/window-manager.js#L15-L33

Which makes me think, that I should probably use robust external polyfill. Let's patch it, and then I can add external dependency in the future.

Cheers!

**Comment by alexandrethsilva** [comment]
@Stanko, no problem. Happy to help with a quick-fix. ✌🏾

**Comment by Stanko** [comment]
I think I found the solution:
```
if `typeof CustomEvent === 'object'`
  // dispatch old school event
else if `typeof CustomEvent !== 'function'`
  // polyfill it
  // dispatch it in a proper way
else
  // do nothing
  // dispatch it in a proper way
```

I'm testing it atm, it works in safari 8 and ie 10. I need to do some more testing and I'll open a PR.

**Comment by Stanko** [comment]
I found this to be the easiest solution to handle all cases.

https://github.com/Stanko/window-scroll-manager/blob/master/index.js#L118-L125

Can you please confirm v1.1.4 is working for you?

EDIT: I updated decorators package as well https://github.com/Stanko/react-window-decorators/commit/62537ef4d3f5c94f4adcdc2530af0619820600da

**Comment by alexandrethsilva** [comment]
Hi @Stanko, sorry for the delay. 1.1.4 is working handsomely.
Thanks a lot for the updates!
